src/main/java/tars/Task.java

```diff
@@ -0,0 +1,196 @@
+package tars;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class represents a generic class with a Description and isDone status
+ * Serves as a base class for other Task types like ToDos, Deadline, Event
+ *
+ * @author csk
+ * @version 1
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructor for new Task with specified description
+     * The task will be initialised with marked as not done
+     *
+     * @param description (description of Task)
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns a status icon depending on the isDone status of the Task
+     *
+     * @return "X" if task is done, " " otherwise returned
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Marks the task as done
+     */
+    public void mark() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as not done
+     */
+    public void unmark() {
+        this.isDone = false;
+    }
+
+    /**
+     * Returns status and description of the task
+     *
+     * @return String Representation of task
+     */
+    @Override
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + this.description;
+    }
+
+    /**
+     * Returns string format of Task to be stored in the text file to retrieved later
+     *
+     * @return String Representation of task
+     */
+    public String toFileFormat() {
+        return "";
+    }
+}
+/**
+* Class represents ToDos task, which only has a description
+* Extends from Task class
+*
+* @see Task
+*/
+class ToDos extends Task {
```

Comment:
this may be a major change so i understand if you wouldnt want to do it xD, but consider renaming this to ToDo (singular) instead. reason is every time you instantiate this class only one instance of a Todo task is created, as opposed to a Collection of ToDo (like a list or something)